packages/react/src/components/dateInput/DateInput.stories.tsx

```diff
@@ -190,5 +190,67 @@ export const WithSelectedDisabledDates = (args) => {
     />
   );
 };
+
 WithSelectedDisabledDates.storyName = 'With selected disabled dates';
 WithSelectedDisabledDates.parameters = { loki: { skip: true } };
+export const WithCustomDayStyles = (args: DateInputProps) => {
+  const dateFormat = 'dd.M.yyyy';
+  const dateValue = new Date(2021, 10, 12);
+  const [value, setValue] = useState<string>(format(dateValue, dateFormat));
+  const hasSomeSpaceLeft = 'some-space-left';
+  const someSpaceColor = 'var(--color-tram-medium-light)';
+  const hasLittleSpaceLeft = 'little-space-left';
+  const littleSpaceColor = 'var(--color-summer-medium-light)';
+  const helperText = `Custom styles for days with limited available timeslots in the date picker calendar. Use format D.M.YYYY.`;
+
+  const dayHasSomeSpace = (day: number) => day === 12 || day === 15 || day === 24 || day === 25;
+  const dayHasLittleSpace = (day: number) => day === 3 || day === 4 || day === 17 || day === 19;
+
+  const setDateClassNames = (date: Date) => {
+    const day = date.getDate();
+    if (dayHasSomeSpace(day)) return `${hasSomeSpaceLeft}--day`;
+    if (dayHasLittleSpace(day)) return `${hasLittleSpaceLeft}--day`;
+    return undefined;
+  };
+
+  const setDateAriaDescribedBy = (date: Date) => {
+    const day = date.getDate();
+    if (dayHasSomeSpace(day)) return hasSomeSpaceLeft;
+    if (dayHasLittleSpace(day)) return hasLittleSpaceLeft;
+    return undefined;
+  };
+
+  return (
+    <>
+      <style>
+        {`
+          .${hasSomeSpaceLeft}--day {
+              background: ${someSpaceColor};
+          }
+
+          .${hasLittleSpaceLeft}--day {
+            background: ${littleSpaceColor};
+          }
+        `}
+      </style>
+      <DateInput
+        {...args}
+        value={value}
+        onChange={setValue}
+        helperText={helperText}
+        setDateClassNames={setDateClassNames}
+        setDateAriaDescribedBy={setDateAriaDescribedBy}
+        legend={[
+          { id: hasSomeSpaceLeft, label: 'Some free timeslots available', color: someSpaceColor },
+          {
```

Comment:
The link between `legend`, `setDateClassNames` and `setDateAriaDescribedBy` should be documented in detail and/or the examples should return legend values from the functions.

I'd hoist the legend props up and replace the code with
```

const someSpaceLeftDate: Legend =   {
  id: hasSomeSpaceLeft, 
  label: 'Some free timeslots available', 
  color: someSpaceColor
}
            
const littleSpaceLeftDate: Legend = {
  id: hasLittleSpaceLeft,
  label: 'Only a few free timeslots available',
  color: littleSpaceColor
 }            
            
const getDateLegendId = (date: Date) => {
 const day = date.getDate();
 if (dayHasSomeSpace(day)) return someSpaceLeftDate.id;
 if (dayHasLittleSpace(day)) return littleSpaceLeftDate.id;
 return undefined
}

const setDateClassNames = (date: Date) => {
   const dateLegendId = getDateLegendId(date)
   return dateLegendId? `${dateLegendId}--day` : undefined;
};

const setDateAriaDescribedBy = (date: Date) => {
   return getDateLegendId(date)
};

const legends=[someSpaceLeftDate, littleSpaceLeftDate]
```

And maybe even

```
const createStyles = ()=>{
  return legends.map(legend=>{
    return .${legend.id}--day {
              background: ${legend.color};
          }
  }).join(" ");
}
```